cheroot/test/test_ssl.py

```diff
@@ -696,6 +694,92 @@ def test_https_over_http_error(http_server, ip_addr):
     assert expected_substring in ssl_err.value.args[-1]
 
 
+@pytest.mark.parametrize(  # noqa: C901  # FIXME
+    'adapter_type',
+    (
+        'builtin',
+        'pyopenssl',
+    ),
+)
+def test_plain_http_check_no_data(
+    adapter_type,
+    tls_certificate_chain_pem_path,
+    tls_certificate_private_key_pem_path,
+    mocker,
+):
+    """Test ``_check_for_plain_http()`` handles empty peek correctly."""
+    tls_adapter_cls = get_ssl_adapter_class(name=adapter_type)
+
+    tls_adapter = tls_adapter_cls(
+        tls_certificate_chain_pem_path,
+        tls_certificate_private_key_pem_path,
+    )
+
+    mock_socket = mocker.MagicMock()
+    mock_socket.recv.return_value = b''  # Empty peek
+    mock_socket.gettimeout.return_value = None
+
+    result = tls_adapter._check_for_plain_http(mock_socket)
+
+    assert result is False
+    mock_socket.recv.assert_called_once_with(16, socket.MSG_PEEK)
+
+
+@pytest.mark.parametrize(
+    'adapter_type',
+    ('builtin', 'pyopenssl'),
+)
+def test_plain_http_check_socket_error(
+    adapter_type,
+    mocker,
+    tls_certificate_chain_pem_path,
+    tls_certificate_private_key_pem_path,
+):
+    """Test ``_check_for_plain_http handles()`` socket errors gracefully."""
+    tls_adapter_cls = get_ssl_adapter_class(name=adapter_type)
+    adapter = tls_adapter_cls(
+        tls_certificate_chain_pem_path,
+        tls_certificate_private_key_pem_path,
+    )
+
+    mock_socket = mocker.MagicMock()
+    mock_socket.recv.side_effect = socket.timeout('Timed out')
+    mock_socket.gettimeout.return_value = None
+
+    result = adapter._check_for_plain_http(mock_socket)
+
+    assert result is False  # Should return False on error
+
+
+@pytest.mark.parametrize(  # noqa: C901  # FIXME
+    'adapter_type',
+    (
+        'builtin',
+        'pyopenssl',
+    ),
+)
+def test_plain_http_check_os_error(
+    adapter_type,
+    mocker,
+    tls_certificate_chain_pem_path,
+    tls_certificate_private_key_pem_path,
+):
+    """Test ``_check_for_plain_http()`` handles OSError gracefully."""
+    tls_adapter_cls = get_ssl_adapter_class(name=adapter_type)
+    adapter = tls_adapter_cls(
+        tls_certificate_chain_pem_path,
+        tls_certificate_private_key_pem_path,
+    )
+
+    mock_socket = mocker.MagicMock()
+    mock_socket.recv.side_effect = OSError('Connection reset')
+    mock_socket.gettimeout.return_value = None
+
+    result = adapter._check_for_plain_http(mock_socket)
+
+    assert result is False
```

Comment:
With the suggested naming, this could read as
```suggestion
    assert not adapter._probe_peer_speaks_http(mock_socket)
```
(or an equivalent `with pytest.raises(...)` which I think I'm starting to prefer)